tests/unit/test_charm.py

```diff
@@ -1,326 +1,153 @@
-    #     mock_boto_client.list_buckets.return_value = []
-    #
-    #     # Ensure the simulated Pebble API is reachable
-    #     self.harness.set_can_connect(CONTAINER, True)
-    #
-    #     relation_id = self.harness.add_relation(S3_INTEGRATOR_REL, "s3-integrator")
-    #     self.harness.add_relation_unit(relation_id, "s3-integrator/0")
-    #     self.harness.update_relation_data(
-    #         relation_id,
-    #         "s3-integrator",
-    #         {
-    #             CONFIG_KEY_S3_ACCESS_KEY: "DUMMY_ACCESS_KEY",
-    #             CONFIG_KEY_S3_SECRET_KEY: "DUMMY_SECRET_KEY",
-    #             CONFIG_KEY_S3_ENDPOINT: "https://dummyendpoint.com",
-    #             CONFIG_KEY_S3_BUCKET: "DUMMY_BUCKET",
-    #             CONFIG_KEY_S3_LOGS_DIR: "DUMMY_LOG_DIR",
-    #         },
-    #     )
-    #
-    #     self.assertEqual(self.harness.model.unit.status, ActiveStatus())
-    #
-    #     self.assertEqual(
-    #         self.harness.charm.spark_config.s3_log_dir, "s3a://DUMMY_BUCKET/DUMMY_LOG_DIR"
-    #     )
-    #
-    #     self.harness.remove_relation(relation_id)
-    #
-    #     self.assertEqual(
-    #         self.harness.model.unit.status, BlockedStatus(STATUS_MSG_MISSING_S3_RELATION)
-    #     )
-    #
-    # @patch("ops.model.Container.exists")
-    # def test_container_failure(self, mock_container_file_exists):
-    #     # mocks
-    #     mock_container_file_exists.return_value = False
-    #     self.harness.set_can_connect(CONTAINER, False)
-    #     self.harness.charm.on.config_changed.emit()
-    #     self.assertEqual(
-    #         self.harness.model.unit.status,
-    #         BlockedStatus(STATUS_MSG_MISSING_S3_RELATION),
-    #     )
-    #
-    #     self.harness.set_can_connect(CONTAINER, True)
-    #
-    #     self.harness.charm.on.config_changed.emit()
-    #
-    #     self.assertEqual(
-    #         self.harness.model.unit.status,
-    #         BlockedStatus(STATUS_MSG_MISSING_S3_RELATION),
-    #     )
+from pathlib import Path
+from unittest.mock import patch
+
+from scenario import Container, State
+
+from constants import CONTAINER
+
+
+def parse_spark_properties(out: State, tmp_path: Path) -> dict[str, str]:
+
+    spark_properties_path = (
+        out.get_container(CONTAINER)
+        .layers["charm"]
+        .services["history-server"]
+        .environment["SPARK_PROPERTIES_FILE"]
+    )
+
+    file_path = tmp_path / Path(spark_properties_path).relative_to("/opt")
+
+    assert file_path.exists()
+
+    with file_path.open("r") as fid:
+        return dict(
+            row.rsplit("=", maxsplit=1) for line in fid.readlines() if (row := line.strip())
+        )
+
+
+def test_start_history_server(history_server_ctx):
+    state = State(
+        config={},
+        containers=[Container(name=CONTAINER, can_connect=False)],
+    )
+    out = history_server_ctx.run("install", state)
+    assert out.unit_status.name == "maintenance"
```

Comment:
you should be able to do `out.unit_status == MaintenanceStatus("Waiting for Pebble")`.